core/modules.rs

```diff
@@ -278,6 +282,75 @@ impl<L: Loader> Future for RecursiveLoad<L> {
   }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+#[allow(dead_code)]
+/// Resolved module specifier
+pub struct ModuleSpecifier(Url);
```

Comment:
I see, so the ModuleSpecifier is a `core` concept instead of a `cli` concept. That's why you have to add the URL dep.

Can we do something more minimal for the first pass:  where `ModuleSpecifier` is only exposed in `//cli` - and `//core` is left unchanged? The reason is that it's already a large change to modify the signatures of `get_source_code()` and `fetch_module_meta_data_async()`. I'd rather see those cleaned up, and if it ends up being a helpful refactor we can consider moving `ModuleSpecifier` into `//core`. What do you think?

Reply:
Sure, I'll rebase this PR, but I'm gonna keep the code from `core/` on separate branch - I spent a few hours rewriting tests for modules, though it's very educational I'd rather not do that again 😅